shadows/framework/src/main/java/org/robolectric/shadows/ShadowNetworkCapabilities.java

```diff
@@ -54,6 +57,28 @@ public NetworkCapabilities removeCapability(int capability) {
         .removeCapability(capability);
   }
 
+  /**
+   * Changes {@link NetworkSpecifier} for this network capabilities. Works only on Android O and
+   * higher. For lower versions use {@link #setNetworkSpecifier(String)}
+   */
+  @HiddenApi
+  @Implementation(minSdk = O)
+  public NetworkCapabilities setNetworkSpecifier(NetworkSpecifier networkSpecifier) {
```

Comment:
From issue description, this method is used for set `NetworkSpecifier` for unit-test. Maybe we don't need to change it to `implementation` method. @hoisie Is that right?

Reply:
In this case you need `@Implementation` because the real NetworkCapabilities has a method `setNetworkSpecifier` that is `@hide`:

https://cs.android.com/android/platform/superproject/+/master:packages/modules/Connectivity/framework/src/android/net/NetworkCapabilities.java;l=1545-1546

The annotation validator may not let you have a method `setNetworkSpecifier` without the `@Implementation`

Reply:
Got it. This is another case to make implementation method public for purpose usage.